hazelcast/src/main/java/com/hazelcast/config/DiscoveryStrategiesConfig.java

```diff
@@ -96,7 +96,7 @@ public boolean isEnabled() {
      * @return all enabled {@link DiscoveryStrategy} configurations
      */
     public Collection<DiscoveryStrategyConfig> getDiscoveryStrategyConfigs() {
-        return Collections.unmodifiableCollection(discoveryStrategyConfigs);
+        return discoveryStrategyConfigs;
```

Comment:
Yes, `DiscoveryConfig` might be better to indicate that this class is top-level config for the  Discovery SPI than the `DiscoveryStrategiesConfig`